buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java

```diff
@@ -0,0 +1,107 @@
+package com.uber.okbuck.core.task;
+
+import com.uber.okbuck.OkBuckGradlePlugin;
+import com.uber.okbuck.core.model.base.ProjectType;
+import com.uber.okbuck.core.util.FileUtil;
+import com.uber.okbuck.core.util.GroovyUtil;
+import com.uber.okbuck.core.util.KotlinUtil;
+import com.uber.okbuck.core.util.ProguardUtil;
+import com.uber.okbuck.core.util.ProjectUtil;
+import com.uber.okbuck.extension.OkBuckExtension;
+import com.uber.okbuck.generator.DotBuckConfigLocalGenerator;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.uber.okbuck.OkBuckGradlePlugin.OKBUCK_DEFS;
+
+public class OkBuckTask extends DefaultTask {
+
+  @Input public OkBuckExtension okBuckExtension;
+
+  @TaskAction
+  void okbuck() {
+    // Fetch Groovy support deps if needed
+    boolean hasGroovyLib = okBuckExtension.buckProjects.parallelStream().anyMatch(project -> ProjectUtil.getType(project) == ProjectType.GROOVY_LIB);
+    if (hasGroovyLib) {
+      GroovyUtil.setupGroovyHome(getProject());
+    }
+
+    // Fetch Kotlin support deps if needed
+    boolean hasKotlinLib = okBuckExtension.buckProjects.parallelStream().anyMatch(project -> ProjectUtil.getType(project) == ProjectType.KOTLIN_LIB);
+    Pair<String, String> kotlinDeps = null;
+    if (hasKotlinLib) {
+      kotlinDeps = KotlinUtil.setupKotlinHome(getProject());
+    }
+
+    generate(okBuckExtension,
+            hasGroovyLib ? GroovyUtil.GROOVY_HOME_LOCATION : null,
+            kotlinDeps != null ? kotlinDeps.getLeft(): null,
+            kotlinDeps != null ? kotlinDeps.getRight(): null);
+  }
+
+  @Override public String getGroup() {
+    return OkBuckGradlePlugin.GROUP;
+  }
+
+  @Override public String getDescription() {
+    return "Setup okbuck cache and dependencies";
```

Comment:
This should be handled by the `setupOkbuck` Task. root okbuck task description should be more like "okbuck task for the root project"